pytorch_lightning/trainer/callback_hook.py

```diff
@@ -42,13 +387,200 @@ def on_train_batch_start(self, batch, batch_idx, dataloader_idx=0):
+        .. deprecated:: v1.6
+            `TrainerCallbackHookMixin.on_test_start` was deprecated in v1.6 and will be removed in v1.7.
+
+        Called when the test begins.
+        """
+        rank_zero_deprecation(
+            "`TrainerCallbackHookMixin.on_test_start` was deprecated in v1.6 and will be removed in v1.7."
+        )
+        for callback in self.callbacks:
+            callback.on_test_start(self, self.lightning_module)
+
+    def on_test_end(self):
+        r"""
+        .. deprecated:: v1.6
+            `TrainerCallbackHookMixin.on_test_end` was deprecated in v1.6 and will be removed in v1.7.
+
+        Called when the test ends.
+        """
+        rank_zero_deprecation(
+            "`TrainerCallbackHookMixin.on_test_end` was deprecated in v1.6 and will be removed in v1.7."
+        )
+        for callback in self.callbacks:
+            callback.on_test_end(self, self.lightning_module)
+
+    def on_predict_start(self) -> None:
+        r"""
+        .. deprecated:: v1.6
+            `TrainerCallbackHookMixin.on_predict_start` was deprecated in v1.6 and will be removed in v1.7.
+
+        Called when predict begins.
+        """
+        rank_zero_deprecation(
+            "`TrainerCallbackHookMixin.on_predict_start` was deprecated in v1.6 and will be removed in v1.7."
+        )
+        for callback in self.callbacks:
+            callback.on_predict_start(self, self.lightning_module)
+
+    def on_predict_end(self) -> None:
+        r"""
+        .. deprecated:: v1.6
+            `TrainerCallbackHookMixin.on_predict_end` was deprecated in v1.6 and will be removed in v1.7.
+
+        Called when predict ends.
+        """
+        rank_zero_deprecation(
+            "`TrainerCallbackHookMixin.on_predict_end` was deprecated in v1.6 and will be removed in v1.7."
+        )
+        for callback in self.callbacks:
+            callback.on_predict_end(self, self.lightning_module)
+
+    def on_keyboard_interrupt(self):
+        r"""
+        .. deprecated:: v1.5
+            This callback hook was deprecated in v1.5 in favor of `on_exception` and will be removed in v1.7.
+
+        Called when any trainer execution is interrupted by KeyboardInterrupt.
+        """
+        for callback in self.callbacks:
+            callback.on_keyboard_interrupt(self, self.lightning_module)
+
+    def on_exception(self, exception: BaseException) -> None:
+        r"""
+        .. deprecated:: v1.6
+            `TrainerCallbackHookMixin.on_exception` was deprecated in v1.6 and will be removed in v1.7.
+
+        Called when any trainer execution is interrupted by an exception.
+        """
+        rank_zero_deprecation(
+            "`TrainerCallbackHookMixin.on_exception` was deprecated in v1.6 and will be removed in v1.7."
+        )
+        for callback in self.callbacks:
+            callback.on_exception(self, self.lightning_module, exception)
+
     def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict[str, dict]:
```

Comment:
are the save/load checkpoint hooks the only ones not deprecated at this point? 

I think `on_save_checkpoint` is used only here in the Trainer: https://github.com/PyTorchLightning/pytorch-lightning/blob/f95976d602d294ea3943dadd6ca4e8f2ed7bab21/pytorch_lightning/trainer/connectors/checkpoint_connector.py#L381

And the naming is *very* confusing: if someone saw `Trainer.on_save_checkpoint` they for sure wouldn't anticipate that it only returns the callback states for callbacks attached to the trainer. especially when compared to `trainer.save_checkpoint` which saves everything. 

I think it's worth discussing (in a separate issue/PR) if we can move these functions to be within the CheckpointConnector so we can deprecate all of `TrainerCallbackHookMixin` and so the definitions here live next to the only place where they're used. 

This really only needs the callbacks, and could be a function like:
```py
get_callback_checkpoint_states(callbacks: Sequence[Callbacks]) -> Dict[str, dict]:
```

Reply:
> are the save/load checkpoint hooks the only ones not deprecated at this point?

yes! Completely agree, was planning on addressing after this PR, but maybe we can address before merging this PR and then this PR can fully deprecate the `TrainerCallbackHookMixin`. Will create an issue